src/flow-timeout.c

```diff
@@ -211,7 +211,7 @@ static inline Packet *FlowForceReassemblyPseudoPacketSetup(
     }
 
     p->tcph->th_offx2 = 0x50;
-    p->tcph->th_flags |= TH_ACK;
```

Comment:
What's interesting to me is that this seems like legacy code and is there in probably all Suricata versions. Could you please tell that despite this flag being there on 6.0.x, why is it not affected?
Thanks a lot!

Reply:
yes indeed, it is in all Suri versions. But for Suri 7 and master it was changed by commit 1bb6f44ff01363fa29488f1ae83b9368e33c2770 which adjusted how are packets evaluated at the shutdown. Beforehand, the packets that didn't reach the application stage reassembly were not flushed by pseudopackets and thus they were not altered by this code path.